site/docs/plugins/conversations.md

```diff
@@ -0,0 +1,900 @@
+  However, when processing thousands of conversations in parallel with different users, it would consume potentially very large amounts of memory.
+  If many users stop responding, this leaves the bot stuck in the middle of countless conversations.
+- Webhooks have their own whole [category of problems](/guide/deployment-types.html#ending-webhook-requests-in-time) with long-running middleware.
+
+**State.**
+On serverless infrastructure such as cloud functions, we cannot actually assume that the same instance handles two subsequent updates from the same user.
+Hence, if we were to create stateful conversations, they may randomly break all the time, as some `wait` calls don't resolve, but some other middleware is suddenly executed.
+The result is an abundance of random bugs and chaos.
+
+There are more problems, but you get the idea.
+
+Consequently, the conversations plugin does things differently.
+Very differently.
+
+The conversations plugin tracks the execution of your function.
+When a wait call is reached, it serializes the state of execution into the session, and safely stores it in a database.
+When the next update arrives, it first inspects the session data.
+If it finds that it left off in the middle of a conversation, it deserializes the state of execution, takes your conversation builder function, and replays it up to the point of the last `wait` call.
+It then resumes ordinary execution of your function—until the next `wait` call is reached, and the execution must be haltet again.
+
+What do we mean by the state of execution?
+In a nutshell, it consists on three things:
+
+1. Incoming updates.
+2. Outgoing API calls.
+3. External events and effects, such as randomness or calls to external APIs or databases.
+
+What do we mean by replaying?
+Replaying simply means calling the function regularly from the start, but when it does things like calling `wait` or performing API calls, we don't actually do any of those.
+Instead, we check out or logs where we recorded from a previous run which values were returned.
+We then inject these values so that the conversation builder function simply runs through very fast---until our logs are exhausted.
+At that point, we switch back to normal execution mode, which is just a fancy way of saying that we stop injecting stuff, and start to actually perform API calls again.
+
+This is why the plugin has to track all incoming updates as well as all Bot API calls.
+(See points 1 and 2 above.)
+However, the plugin has no control over external events, side-effects, or randomness.
+For example, you could this:
+
+```ts
+if (Math.random() < 0.5) {
+  // do one thing
+} else {
+  // do another thing
+}
+```
+
+In that case, when calling the function, it may suddenly behave differently every time, so replaying the function will break!
+It could randomly work differently than the original execution.
+This is why point 3 exists, and [The Three Golden Rules](#three-golden-rules-of-conversations) must be followed.
```

Comment:
```suggestion
This is why point 3 exists, and the [Three Golden Rules](#three-golden-rules-of-conversations) must be followed.
```

Reply:
Do you want to see this applied across the entire website?

Reply:
I think it is already like that, at least places where I have modified

Reply:
I did a quick case-insensitive search for `[the` and it seems like there are dozens of places that include the _the_ into the link.

Reply:
That doesn't mean you can't commit this suggestion for now.

Reply:
I will rather fix all other places in the same file with the same commit, and open a new PQ that fixes everything else.